FitBitApi/FitBitApi/AppDelegate.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  AppDelegate.swift
+//  FitBitApi
+//
+//  Created by Wishell on 13.11.2018.
+//  Copyright © 2018 Wishell. All rights reserved.
+//
+
+import UIKit
+import KeychainAccess
+
+@UIApplicationMain
+class AppDelegate: UIResponder, UIApplicationDelegate {
+
+    var window: UIWindow?
+
+
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        // Override point for customization after application launch.
+        return true
+    }
+
+    func applicationWillResignActive(_ application: UIApplication) {
+        // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
+        // Use this method to pause ongoing tasks, disable timers, and invalidate graphics rendering callbacks. Games should use this method to pause the game.
+    }
+
+    func applicationDidEnterBackground(_ application: UIApplication) {
+        // Use this method to release shared resources, save user data, invalidate timers, and store enough application state information to restore your application to its current state in case it is terminated later.
+        // If your application supports background execution, this method is called instead of applicationWillTerminate: when the user quits.
+    }
+
+    func applicationWillEnterForeground(_ application: UIApplication) {
+        // Called as part of the transition from the background to the active state; here you can undo many of the changes made on entering the background.
+    }
+
+    func applicationDidBecomeActive(_ application: UIApplication) {
+        // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
+    }
+
+    func applicationWillTerminate(_ application: UIApplication) {
+        // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
+    }
+    
+    func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any] = [:]) -> Bool {
+        let oauthManager = OauthManager()
+        oauthManager.handle(url) // TODO error handling
+
+        let storyboard : UIStoryboard = UIStoryboard(name: "Main", bundle: nil)
+        
+        let vc: PresenterViewController = storyboard.instantiateViewController(withIdentifier: "PresenterViewController") as! PresenterViewController
+        self.window?.rootViewController = vc
```

Comment:
Use NotificationCenter to post notification about success login here
https://developer.apple.com/documentation/foundation/nsnotificationcenter

Subscribe to posted event in Login controller and make segue to the Presenter

declare Notification.Name extension

```
extension Notification.Name {
     static let successLogin = Notification.Name("successLogin")
}
```

remove code

```
let storyboard : UIStoryboard = UIStoryboard(name: "Main", bundle: nil)
        
        let vc: PresenterViewController = storyboard.instantiateViewController(withIdentifier: "PresenterViewController") as! PresenterViewController
        self.window?.rootViewController = vc
```

Add next code

`NotificationCenter.default.post(name: .successLogin, object: nil)`